CPT331.iOS/CPT331.iOS/Event.swift

```diff
@@ -9,17 +9,22 @@
 import Foundation
 import CoreLocation
 
-struct Event {
+class Event {
```

Comment:
Perfect. This should improve performance and memory usage. 

We'll only need to pass a single reference rather than the entire struct.